### PR TITLE
File io utility

### PR DESCRIPTION
**Pull request by michaelni678**
added rezasm file struct
seek and peek, get bytes, etc
simple tests for all of those

### CODE REVIEW

**Comment by michaelni678** [comment]
Closes #10

File: rezasm-source/rezasm-core/src/util/error.rs [hunk 1]
```diff
@@ -114,6 +117,21 @@ pub enum SimulatorError {
 
     #[error("label `{0}` is already in use")]
     LabelInUseError(String),
+
+    #[error("io failure")]
```
Comment (WyattRoss) [inline rezasm-source/rezasm-core/src/util/error.rs:121]: Output the string for more detail

File: rezasm-source/rezasm-core/src/util/error.rs [hunk 2]
```diff
@@ -25,6 +25,9 @@ pub enum EzasmError {
     #[error("file `{0}` does not exist")]
     FileDoesNotExistError(String),
 
+    #[error("directory does not exist")]
```
Comment (WyattRoss) [inline rezasm-source/rezasm-core/src/util/error.rs:28]: Unused: remove?

File: rezasm-app/rezasm-cli/src/main.rs [hunk 3]
```diff
@@ -226,3 +227,58 @@ pub fn test_simulator_labels() {
...
+
+    // Convert to a writer.
+    let path = concat!(env!("CARGO_MANIFEST_DIR"), "/example/arith_modified.ez");
+    let mut writer = rezasmfile.into_writer(Some(path)).unwrap();
+    let write_str = "\n# This comment was generated by io test!";
+    writer.extend_from_slice(write_str.as_bytes());
+    writer.flush().unwrap();
+
+    // Make sure file exists.
+    assert_eq!(fs::metadata(path).is_err(), true);
```
Comment (WyattRoss) [inline rezasm-app/rezasm-cli/src/main.rs:281]: Platform specific & unnecessary. Remove

File: rezasm-source/rezasm-core/src/util/io.rs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    /// Return the byte at the current position and then advance the cursor forward by one.
+    /// Returns none if out of bounds.
+    pub fn next(&mut self) -> Option<u8> {
+        self.cursor += 1;
+        self.peek_relative_byte(-1).ok()
+    }
+    /// Return the byte at the current position and then advance the cursor backward by one.
+    /// Returns none if out of bounds.
+    pub fn prev(&mut self) -> Option<u8> {
```
Comment (WyattRoss) [inline rezasm-source/rezasm-core/src/util/io.rs:67]: check to see if cursor is already 0

**Review by WyattRoss — CHANGES_REQUESTED** [review]
See comments.

**Pull request merged**